lib/charms/opensearch/v0/opensearch_nodes_exclusions.py

```diff
@@ -41,76 +42,196 @@ class OpenSearchExclusions:
     def __init__(self, charm):
         self._charm = charm
         self._opensearch = self._charm.opensearch
-
         self._scope = Scope.APP if self._charm.unit.is_leader() else Scope.UNIT
 
-    def add_current(self, restart: bool = False) -> None:
+    def add(self, unit_name: str) -> None:
```

Comment:
I'm adding the question here though it should go rather on line 45 -- impacting the whole module.

I gather that we do not save anything in the databag on the leader unit, as leader only manipulates the `app` databag.

Now, if a leader is restarted, and leadership may be granted to another unit, then the ex-leader comes back with an empty unit databag (in terms of voting exclusions).

Why is that not an issue...?

Reply:
So, the `VOTING_TO_DELETE` only matters at `cleanup()`. At that moment, we will do the following:
1) Load all the `VOTING_TO_DELETE` entries from our unit/app databag (depending if we are the leader or not)
2) For each entry in that set, we only will clean it from "voting exclusion" IF that unit is gone from juju
3) We apply the final cleanup

At restart, we call: `add_current()` at `_stop_opensearch`, i.e. we will forcely add our own unit to the exclusions. We then stop and start the service. At `_start_opensearch`, we call `delete_current()`, which will remove this same unit from the exclusion, forcely, and then update its own databag.

Therefore, my changes only really take effect on: (1) `-departed`, the app leader registers the new unit to be removed; and (2) at cleanup: we remove the unit from the voting exclusion if the unit is not available on Juju anymore.